### PR TITLE
refactor(tsconfig): Move datasources, managers and versioning to excluded section

### LINKED ISSUE #13971: Use explicit list of files excluded from strict null checks
### Describe the proposed change(s).

Explicit list of files that we have to fix in terms of null checks is long.
However, it's more straightforward to maintain: once some file is fixed, you just remove a single line.
Also, we can see the overall progress by looking at `tsconfig.strict.json` LOC count.

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

- Include `lib/` and `test/`
- Exclude all `*.spec.ts` for now
- List all `*.ts` files

## Context:

- Closes: #13971

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: tsconfig.strict.json [hunk 1]
```diff
@@ -5,101 +5,499 @@
...
-    "test/newline-snapshot-serializer.ts",
-    "test/static-files.spec.ts",
-    "tools/utils/index.ts"
-  ],
+  "include": ["lib/", "test/"],
   "exclude": [
-    "lib/constants/platform.spec.ts",
-    "lib/datasource/adoptium-java/**/*.spec.ts",
-    "lib/datasource/artifactory/**/*.spec.ts",
+    "lib/**/*.spec.ts",
```
Comment (zharinov) [inline tsconfig.strict.json:10]: Let's ignore tests until source files are fixed.

Lots of tests are using `index.ts` files from the top-level. It causes circular behaviour which is a little bit overwhelming when you're trying to quickly fix a single folder.

File: tsconfig.strict.json [hunk 2]
```diff
@@ -5,101 +5,499 @@
...
-    "lib/workers/repository/model/commit-message.ts",
-    "test/exec-util.ts",
-    "test/graphql-snapshot.ts",
-    "test/http-mock.ts",
-    "test/json-schema.ts",
-    "test/newline-snapshot-serializer.ts",
-    "test/static-files.spec.ts",
-    "tools/utils/index.ts"
-  ],
+  "include": ["lib/", "test/"],
```
Comment (viceice) [inline tsconfig.strict.json:8]: Can we remove this an reuse from base? Like to have it for all files.

**Review by viceice — COMMENTED** [review]
Otherwise LGTM

**Review by viceice — COMMENTED** [review]
needs deconflicting

**Comment by zharinov** [comment]
Let's finish with `maven` and `hex` first?

**Comment by viceice** [comment]
> Let's finish with `maven` and `hex` first?

As you like 🤷‍♂️

**Review by viceice — APPROVED** [review]

**Pull request merged**